wiki/Upgrading-Target-Sdk-Guide.md

```diff
@@ -0,0 +1,148 @@
+- If the breakage does not occur on the non-upgraded version of the app, file a new bug in the issue tracker and mention that it's specific to the new target SDK version and is a blocking issue.
+
+Please note all found blocking issues with their issue numbers in a follow-up reply to the tracking issue. Any other thoughts or findings during testing can also be noted in the tracking issue (similar to the comments in #5137). Please also note in the tracking issue when testing has concluded.
+
+### Part 4: Fixing and stabilizing support
+
+Work on fixing all identified problems from parts (3) and (4) (either by directly fixing the problems via code changes, or via coordination with other members of the team).
+
+Once all issues are fixed, verify each problem is correctly addressed using a temporary local build of the app (see part (3) above). If any problems are still occurring, reopen the corresponding tracking issue and leave a follow-up comment detailing the ongoing problem and steps to reproduce it.
+
+Once all fixed issues are verified, leave a follow-up comment on the upgrade target SDK tracking issue mentioning that verification has concluded and there are no remaining issues found.
+
+### Part 5: Submitting the upgrade
+
+The actual code change to upgrade the app comes in two parts:
+1. Upgrading the compile version (which may already be done as the team sometimes needs to update this for other reasons).
+2. Upgrading the runtime target SDK version.
+
+The sub-sections below detail each of the code changes needed to perform these upgrades.
+
+**Important caveats and notes**:
+- Both version upgrades can be done together in the same PR, but if they are split up the compile-time change (part (5.1) below) must happen first.
+- #5222 is an example of a PR that performs both steps in one, though it includes a few additional code changes that were needed as a result of the compile SDK change.
+- Please note the CI results for this upgrade change. Any failures are likely problems that will need to be fixed within the upgrade PR (if small, e.g. the change in #5222) or filed as a separate bug that will need to be fixed before the upgrade can be submitted (see part (4) above).
+- Updating the compile-time SDK version may require updating the build tools version. This is **not** a simple change and may cause difficult-to-fix breakages due to subtle compatibility issues between third-party dependencies and the build system configurations. If you suspect a build tools version upgrade is needed, please file an issue to track it and contact the developer workflow team lead to discuss next steps.
+- Updating tests to use a newer version of Android can be exceptionally complicated, and thus this is considered a completely separate exercise from upgrading production code. See part (6) below for more specifics.
+
+#### Part 5.1: Updating the compiled SDK version
```

Comment:
The goal is the latter--we don't actually want to create and submit an upgrade PR until it's time for that to be merged (which is when auditing and testing has completed). I was also a bit concerned with this organizational approach since it leads to lots of jumping around. Some possible options we have:
- Keep it as-is since the overall process is fairly short.
- Introduce an appendix with sections detailing specific tasks that need to be done and how, and update the process to point to these steps (this keeps the process itself linear but still results in jumping to/from the appendix, maybe even more than the current writing).
- Split this into two sections as you have it (though my concern with that is 'how to upgrade' doesn't actually have as strong of a completion indication compared to the other sections).

I slightly lean toward (1) since the other two _seem_ more complicated, but I'm not sold on any of the approaches. Do you have any thoughts?

Reply:
Makes sense. I was considering whether that might create a gap in the flow..., but as you pointed out, sticking with option 1 seems like the better approach (for now).